src/store/modules/purchaseOrder/actions.ts

```diff
@@ -11,19 +11,23 @@ import emitter from '@/event-bus'
 const actions: ActionTree<PurchaseOrderState, RootState> = {
 
   // Find Product
-  async findPurchaseOrders ({ commit, state }) {
+  async findPurchaseOrders ({ commit, state }, payload) {
 
     // Show loader only when new query and not the infinite scroll
     // if (payload.viewIndex === 0) emitter.emit("presentLoader");
 
     let resp;
 
     try {
-      resp = await PurchaseOrdersService.fetchPurchaseOrders()
+      resp = await PurchaseOrdersService.fetchPurchaseOrders(payload)
 
       // resp.data.response.numFound tells the number of items in the response
+      const orders = resp.data.grouped.orderId.groups
       if (resp.status === 200 && !hasError(resp)) {
```

Comment:
If the resp.status is 200 but the data is empty then the state will not be updated as expected there might be an error. So add a condition to check whether we are getting some data with status 200 or not

Reply:
Improved code Sir.